coralproject/talk
# Reject reasons

## What does this PR do?
This PR implements behaviors related to the persistence of Moderation Reasos in accordance with the DSA. It defines types for modeling moderation reasons, and updates backend mutations to require them when DSA features are enabled. On the frontend, it provides UI components for indicating these reasons when rejecting a comment.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?
The ModerationReasonCode enum and ModerationReason type are defined, and now included in CommentModerationActions, DSAReportDecisions, and anywhere else the are relevant. 

## Does this PR introduce any new environment variables or feature flags?
No.

## If any indexes were added, were they added to `INDEXES.md`?
n/a

## How do I test this PR?
1. With DSA features *disabled*, verify that moderating comments works as usual
2. Enable DSA features.
3. In the admin, click to reject a comment in the queue. Observe that a modal opens are requires you to indicate the reason for the rejection. For each reason, you should be able to provide addition details.
4. If the reason is "Other" you should be *required* to provide extra details.
5. Once you have filled out this modal, click "Reject" and observe that the comment is rejected.
6. Look in the DB for CommentModerationActions related to that comment, observe that the reason has been persisted on that record.

## Where any tests migrated to React Testing Library?
No.


## How do we deploy this PR?
No special considerations should be required.
